src/App.tsx

```diff
@@ -1,27 +1,78 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum Filter {
+  All = 'all',
+  Five = 'five',
+  Red = 'red',
+  None = '',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+function preparedGoods(filter: Filter): Promise<Good[]> | null {
+  switch (filter) {
+    case Filter.All:
+      return getAll();
+    case Filter.Five:
+      return get5First();
+    case Filter.Red:
+      return getRedGoods();
+    default:
+      return null;
+  }
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.None);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  useEffect(() => {
+    preparedGoods(filter)?.then(response => setGoods(response));
+  }, [filter]);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  function getAllGoods() {
+    setFilter(Filter.All);
+  }
```

Comment:
try to remove `filter` state variable and use `useCallback` for each click function